datalad/interface/add_archive_content.py

```diff
@@ -78,87 +77,107 @@
+        dict(text="""Add files from the archive 'tarball.tar.gz', and
+                     remove big_tarball.tar.gz from the index""",
+             code_py="add_archive_content(path='big_tarball.tar.gz', delete=True)",
+             code_cmd="datalad add-archive-content big_tarball.tar.gz --delete"),
+        dict(text="""Add files from the archive 's3.zip' but remove the leading
+                     directory""",
+             code_py="add_archive_content(path='s3.zip', strip_leading_dirs=True)",
+             code_cmd="datalad add-archive-content s3.zip --strip-leading-dirs"),
+        ]
+
     # XXX prevent common args from being added to the docstring
     _no_eval_results = True
     _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc=""""specify the dataset to save""",
+            constraints=EnsureDataset() | EnsureNone()),
         delete=Parameter(
-            args=("-d", "--delete"),
+            args=("-D", "--delete"),
             action="store_true",
-            doc="""flag to delete original archive from the filesystem/git in current tree.
-                   %s""" % _KEY_OPT_NOTE),
+            doc="""delete original archive from the filesystem/Git in current
+            tree. %s""" % _KEY_OPT_NOTE),
         add_archive_leading_dir=Parameter(
             args=("--add-archive-leading-dir",),
             action="store_true",
-            doc="""flag to place extracted content under a directory which would correspond
-                   to archive name with suffix stripped.  E.g. for archive `example.zip` its
-                   content will be extracted under a directory `example/`"""),
+            doc="""place extracted content under a directory which would
+            correspond to the archive name with suffix stripped. E.g. the
+            content of `example.zip` will be extracted under `example/`"""),
         strip_leading_dirs=Parameter(
             args=("--strip-leading-dirs",),
             action="store_true",
-            doc="""flag to move all files directories up, from how they were stored in an archive,
-                   if that one contained a number (possibly more than 1 down) single leading
-                   directories"""),
+            doc="""move all archive contents up from how they are stored in the
+             archive, if the archive contains a number (possibly more than 1
+             down) single leading directories"""),
         leading_dirs_depth=Parameter(
             args=("--leading-dirs-depth",),
             action="store",
             type=int,
-            doc="""maximal depth to strip leading directories to.  If not specified (None), no limit"""),
+            doc="""maximal depth to strip leading directories to.
+            If not specified (None), no limit"""),
         leading_dirs_consider=Parameter(
             args=("--leading-dirs-consider",),
```

Comment:
Maybe transition argnames?

```suggestion
            args=("--leading-dirs-consider", "--leading-dirs-regex"),
```

Reply:
Ha, I didn't know this was possible. lets discuss alternatives at todays call?